packages/contracts-periphery/test/contracts/universal/Transactor.spec.ts

```diff
@@ -0,0 +1,116 @@
+import hre from 'hardhat'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import { Contract } from 'ethers'
+
+import { expect } from '../../setup'
+import { decodeSolidityRevert, deploy } from '../../helpers'
+
+describe('AssetReceiver', () => {
+  let signer1: SignerWithAddress
+  let signer2: SignerWithAddress
+  before('signer setup', async () => {
+    ;[signer1, signer2] = await hre.ethers.getSigners()
+  })
+
+  let CallRecorder: Contract
+  let Reverter: Contract
+  let Transactor: Contract
+  beforeEach('deploy contracts', async () => {
+    CallRecorder = await deploy('CallRecorder')
+    Reverter = await deploy('Reverter')
+    Transactor = await deploy('Transactor', {
+      signer: signer1,
+      args: [signer1.address],
+    })
+  })
+
+  describe('CALL', () => {
+    describe('when called by authorized address', () => {
+      it('should do a call to the target contract', async () => {
+        const data = CallRecorder.interface.encodeFunctionData('record')
+        await Transactor.CALL(CallRecorder.address, data, 1_000_000, 0, {
+          gasLimit: 2_000_000,
+        })
+
+        const call = await CallRecorder.lastCall()
```

Comment:
couldn't we do this with smock?

Reply:
Nah callvalue not recorded in smock yet, altho I might fix that later